### PR TITLE
Add YouTubeVideo and VideoTutorial component

### LINKED ISSUE #214: Implement VideoTutorial component
- Implement YouTubeVideo component.
- Implement VideoTutorial component.

Part 4 of the implementation of pull #203

### PR DESCRIPTION
**Pull request by IreneStr**
## Summary

This PR can be summarized in the following changelog entry:

* Adds the YouTubeVideo and VideoTutorial component

## Test instructions

Paste this into ContentAnalysis.js
````js
import React from "react";
import styled from "styled-components";

import VideoTutorial from "../../../HelpCenter/views/VideoTutorial";

export const ContentAnalysisContainer = styled.div`
	box-sizing: border-box;
	padding: 16px;
	min-height: 700px;
	width: 100%;
	background-color: white;
`;

const items= [
	{
		title: "Need some help?",
		description: "Go Premium and our experts will be there for you to answer any questions you might have about the setup and use of the plugin.",
		link: "#1",
		linkText: "Get Yoast SEO Premium now »",
	},
	{
		title: "Want to be a Yoast SEO Expert?",
		description: "Follow our Yoast SEO for WordPress training and become a certified Yoast SEO Expert!",
		link: "#2",
		linkText: "Enroll in the Yoast SEO for WordPress training »",
	},
];

/**
 * Returns the ContentAnalysis component.
 *
 * @returns {ReactElement} The ContentAnalysis component.
 */
export default function ContentAnalysis() {
	return <ContentAnalysisContainer>
		<VideoTutorial
			src="https://www.youtube.com/embed/bIgcj_pPIbw"
			title="Video title"
			items={ items }/>
	</ContentAnalysisContainer>;
}
````

[Testing code edited by Alexander and Andrea]

Fixes #214

### CODE REVIEW

File: composites/HelpCenter/views/VideoTutorial.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * @param {Object} props The props to use for the component.
+ * @returns {ReactElement} The rendered VideoTutorial component.
+ *
+ * @constructor
+ */
+export default function VideoTutorial( props ) {
+	return (
+		<VideoTutorialPlaceholder>
+			<StyledYouTubeVideo	src={ props.src } title={ props.title } />
+			<VideoTextPanel>
```
Comment (Xyfi) [inline composites/HelpCenter/views/VideoTutorial.js]: Don't use sections of it's only for styling purposes.
https://developer.mozilla.org/nl/docs/Web/HTML/Element/section

File: composites/HelpCenter/views/VideoTutorial.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * Creates a VideoTutorial component that displays a YouTube video.
+ *
+ * @param {Object} props The props to use for the component.
+ * @returns {ReactElement} The rendered VideoTutorial component.
+ *
+ * @constructor
+ */
+export default function VideoTutorial( props ) {
+	return (
+		<VideoTutorialPlaceholder>
```
Comment (Xyfi) [inline composites/HelpCenter/views/VideoTutorial.js]: Sections should be identified by a header (e.g. `h2`).
https://developer.mozilla.org/nl/docs/Web/HTML/Element/section

Maybe create an invisible header for a11y reasons. We could use https://github.com/Yoast/yoast-components/issues/278 for that, an idea of @afercia.

File: composites/HelpCenter/views/VideoTutorial.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	flex: 50% 0;
+`;
+
+/**
+ * Creates a VideoTutorial component that displays a YouTube video.
+ *
+ * @param {Object} props The props to use for the component.
+ * @returns {ReactElement} The rendered VideoTutorial component.
+ *
+ * @constructor
```
Comment (Xyfi) [inline composites/HelpCenter/views/VideoTutorial.js]: Should be removed.

File: composites/HelpCenter/views/VideoTutorial.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+const StyledYouTubeVideo = styled( YouTubeVideo )`
+	flex: 50% 0;
+`;
+
+/**
+ * Creates a VideoTutorial component that displays a YouTube video.
+ *
+ * @param {Object} props The props to use for the component.
+ * @returns {ReactElement} The rendered VideoTutorial component.
```
Comment (Xyfi) [inline composites/HelpCenter/views/VideoTutorial.js]: `rendered` is superfluous.

File: composites/HelpCenter/views/tests/VideoTutorialTest.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import renderer from "react-test-renderer";
+
+import VideoTutorial from "../VideoTutorial.js";
+
+test( "the VideoTutorial component matches the snapshot", () => {
+	const component = renderer.create(
+		<VideoTutorial src="https:/www.youtube.com/embed/bIgcj_pPIbw" title="Video title" description="Video description"/>
```
Comment (Xyfi) [inline composites/HelpCenter/views/tests/VideoTutorialTest.js]: Each prop should be on a separate line.

File: composites/HelpCenter/views/VideoTutorial.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+`;
+
+const StyledYouTubeVideo = styled( YouTubeVideo )`
+	flex: 50% 0;
+`;
+
+/**
+ * Creates a VideoTutorial component that displays a YouTube video.
+ *
+ * @param {Object} props The props to use for the component.
```
Comment (Xyfi) [inline composites/HelpCenter/views/VideoTutorial.js]: New line between @param en @returns.

File: composites/HelpCenter/views/VideoTutorial.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+ * @returns {ReactElement} The rendered VideoTutorial component.
+ *
+ * @constructor
+ */
+export default function VideoTutorial( props ) {
+	return (
+		<VideoTutorialPlaceholder>
+			<StyledYouTubeVideo	src={ props.src } title={ props.title } />
```
Comment (Xyfi) [inline composites/HelpCenter/views/VideoTutorial.js]: Where there is more than 1 prop, place each on a new line.

File: composites/basic/YouTubeVideo.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+const StyledIFrame = styled( IFrame )``;
+
+/**
+ * Creates a YouTubeVideo component.
+ *
+ * @param {Object} props The props to use for the component.
+ *
+ * @returns {ReactElement} The YouTubeVideo component.
+ *
+ * @constructor
```
Comment (Xyfi) [inline composites/basic/YouTubeVideo.js]: Should be removed.

File: composites/basic/tests/YouTubeVideoTest.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import renderer from "react-test-renderer";
+
+import YouTubeVideo from "../YouTubeVideo.js";
+
+test( "the YouTubeVideo component matches the snapshot", () => {
+	const component = renderer.create(
+		<YouTubeVideo src="https:/www.youtube.com/embed/bIgcj_pPIbw" title="Video title"/>
```
Comment (Xyfi) [inline composites/basic/tests/YouTubeVideoTest.js]: Each prop on a new line.

File: composites/basic/YouTubeVideo.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+ */
+export default function YouTubeVideo( props ) {
+	return (
+		<StyledIFrame
+			width={ props.width }
+			height={ props.height }
+			src={ props.src }
+			title={ props.title }
+		    frameBorder={ props.frameBorder }
```
Comment (Xyfi) [inline composites/basic/YouTubeVideo.js]: Spaces should be tabs.

File: composites/basic/YouTubeVideo.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+		    allowFullScreen={ props.allowFullScreen }
```
Comment (Xyfi) [inline composites/basic/YouTubeVideo.js]: Spaces should be tabs.

File: composites/HelpCenter/views/VideoTutorial.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
```
Comment (Xyfi) [inline composites/HelpCenter/views/VideoTutorial.js]: Separate relative and package imports.

File: composites/basic/YouTubeVideo.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import IFrame from "../../composites/basic/IFrame";
```
Comment (Xyfi) [inline composites/basic/YouTubeVideo.js]: Empty line between relative and package imports.

File: composites/basic/YouTubeVideo.js [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+const StyledIFrame = styled( IFrame )``;
```
Comment (Xyfi) [inline composites/basic/YouTubeVideo.js]: Seems unnecessary.

**Comment by Xyfi** [comment]
CR :construction:

**Comment by afercia** [comment]
Not sure a section element is a proper choice also for the container. This component is just a video with a description, not exactly a "thematic grouping of content".

https://www.w3.org/TR/html51/sections.html#the-section-element
> The section element represents a generic section of a document or application. A section, in this context, is a thematic grouping of content. Each section should be identified, typically by including a heading (h1-h6 element) as a child of the section element.

**Comment by afercia** [comment]
Current rendering:

screen shot 2017-09-20 at 12 37 23

**Comment by Xyfi** [comment]
For reference: https://github.com/Yoast/yoast-components/issues/197

**Comment by afercia** [comment]
This in the testing code prevented IE11 and Edge to display the video:
`src="https:/www.youtube.com/embed/bIgcj_pPIbw"`

just add the missing slash:
`src="https://www.youtube.com/embed/bIgcj_pPIbw"`

and it works. 

There's a JS warning
`Warning: Invalid argument supplied to oneOfType, expected an instance of array.`

related to `makeOutboundLink` where `oneOfType()` should use an array.

**Comment by afercia** [comment]
CR 👍 
applied some CS
removed unstyled styled components to avoid empty CSS rules being output
fixed the `oneOfType()` syntax in `makeOutboundLink`

**Comment by IreneStr** [comment]
Acceptance 👍

**Pull request merged**